js.go

```diff
@@ -422,11 +433,23 @@ func (js *js) PublishMsg(m *Msg, opts ...PubOpt) (*PubAck, error) {
 	}
 
 	if err != nil {
-		if err == ErrNoResponders {
-			err = ErrNoStreamResponse
+		for r := 0; err == ErrNoResponders && r < o.rnum; r++ {
```

Comment:
```suggestion
		for r := 0; err == ErrNoResponders && (r < o.rnum || o.rnum < 0); r++ {
```

That change would make it possible to allow infinite retries, in order to let retries happen as needed while context is still active:

```go
ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
defer cancel()
_, err := js.Publish(subj, msg, nats.Context(ctx), nats.RetryWait(500*time.Millisecond), nats.RetryAttempts(-1))
```

Reply:
Should we make sure a ctx is present to do that one?
I will make the change but we could get into an infinite loop if the stream is truly gone and their is no context provided to the Publish call.

Reply:
Change made below.